### ISSUE TITLE
Please upgrade OData to Microsoft.AspNet.WebApi.Core v 5.2.3

### ISSUE THREAD

**Issue by Lingxi-Li** [OData/WebApi#781]
Forwarded from https://github.com/OData/odata.net/issues/649

**Comment by xuzhg** [comment]
Hi,  @jhensley707 

I follow you steps and can't find the issue. Below is what I have done:

1. Use VS2013 to create a new empty web application. select [ASP.NET Empty Web Application]
2. Right click on the created project, select [Add] -> [New Scaffolded Item ...]
3. In the [Add Scaffold] Dialog, select [Web API 2 Controller - Empty]
4. Accept the default controller name and click [Add] button

5. In Package Manager Console, type `Install-Package Microsoft.AspNet.OData` to install Web API OData (5.9.1).

6. In Package Manager Console, type `Install-Package Microsoft.AspNet.WebApi.Cors ` to install Cors (5.2.3).

7. Change WebApiConfig.cs as:

```C#
public static void Register(HttpConfiguration config)
{
     config.MapODataServiceRoute("odata", "odata", GetEdmModel());
     config.EnableCors();
 }
private static IEdmModel GetEdmModel()
{
       var builder = new ODataConventionModelBuilder();
        builder.EntitySet<Customer>("Customers");
        return builder.GetEdmModel();
 }
public class Customer
{
     public int Id { get; set; }
     public string Name { get; set; }
 }

```

8 . Change DefaultController as:

```C#
public class CustomersController : ODataController
{
    public IHttpActionResult Get()
    {
         return Ok(new[] {new Customer {Id = 1, Name = "Sam"}});
     }
 }
```

9 .  Run the application, I can get:

![image](https://cloud.githubusercontent.com/assets/9426627/17282964/f2e95f74-57dd-11e6-9e05-c6dcd87cc854.png)


10 . My 'Web.config' has the following items:

```xml
   <dependentAssembly>
        <assemblyIdentity name="System.Net.Http.Formatting" publicKeyToken="31bf3856ad364e35" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-5.2.3.0" newVersion="5.2.3.0" />
      </dependentAssembly>
```

Would you please compare them with your project? Please let us know the result. 
Thanks.

**Comment by jhensley707** [comment]
According to packages.config, my web service originally had this version of OData
`<package id="Microsoft.AspNet.OData" version="5.2.1" targetFramework="net45" />`
and my web.config specified these assemblies
`<dependentAssembly>`
`<assemblyIdentity name="System.Net.Http.Formatting" publicKeyToken="31bf3856ad364e35" culture="neutral" />`
`<bindingRedirect oldVersion="0.0.0.0-5.2.2.0" newVersion="5.2.2.0" />`
`</dependentAssembly>`
`<dependentAssembly>`
`<assemblyIdentity name="Microsoft.Data.OData" publicKeyToken="31bf3856ad364e35" culture="neutral" />`
`<bindingRedirect oldVersion="0.0.0.0-5.6.2.0" newVersion="5.6.2.0" />`
`</dependentAssembly>`
`<dependentAssembly>`
`<assemblyIdentity name="Microsoft.Data.Edm" publicKeyToken="31bf3856ad364e35" culture="neutral" />`
`<bindingRedirect oldVersion="0.0.0.0-5.6.2.0" newVersion="5.6.2.0" />`
`</dependentAssembly>`

When I used Manage NuGet Packages for the solution and added Cors, it upgraded System.Net.Http.Formatting to v5.2.3. That generated the compatibility error, indicating that System.Net.Http.Formatting v5.2.2 could not be found. The assembly referencing it was OData.

When I upgraded OData to the latest version (package Microsoft ASP.NET Web API 2.2 for OData v4.0), Microsoft.AspNet.OData v5.9.1, it indicated a dependency on Microsoft.AspNet.WebApi.Core >= v5.2.2, and continued to generate the same run-time error. At that time, we removed OData (which was not actually being implemented) and the solution worked fine.

Based on your input, I RESET my solution back one commit to the original state and tried to duplicate the original issue by installing Cors on the web service with the old package of OData. It seems to work fine now. The version of System.Net.Http.Formatting in the bin folder is v5.2.3 and the service works. I can't explain the original issue. It was simply endlessly frustrating.
